File: backend/server/src/main/java/rs/ac/uns/ftn/sbnz/service/PropertyService.java
package rs.ac.uns.ftn.sbnz.service;

import rs.ac.uns.ftn.sbnz.models.Property;
import rs.ac.uns.ftn.sbnz.models.drools.SmartSearch;
import rs.ac.uns.ftn.sbnz.models.drools.ScoredProperties;
import rs.ac.uns.ftn.sbnz.models.enums.PropertyStatus;

import java.util.List;
import java.util.Optional;

public interface PropertyService {

    void addProperty(Property property);

    List<Property> getProperties();

    List<Property> getProperties(PropertyStatus status);

    ScoredProperties getOptimalProperties(SmartSearch smartSearch);

    Property recommend(Property property);

    Optional<Property> getProperty(Long id);

    void updateProperty(Long id, Property property);

    void removeProperty(Long id);
}

File: frontend/src/services/PropertyService.js
import axios from "axios";

export const PropertyService = {
  getAll,
  get,
  addProperty,
  updateProperty,
  priceRecommendation,
  findOptimal,
};

async function getAll(status) {
  if (status)
    return await axios.get(`${process.env.REACT_APP_API_URL}/v1/properties?status=${status}`);
  return await axios.get(`${process.env.REACT_APP_API_URL}/v1/properties`);
}

async function get(id) {
  return await axios.get(`${process.env.REACT_APP_API_URL}/v1/properties/${id}`);
}

async function addProperty(property) {
  return await axios.post(`${process.env.REACT_APP_API_URL}/v1/properties`, property);
}

async function updateProperty(id, property) {
  return await axios.put(`${process.env.REACT_APP_API_URL}/v1/properties/${id}`, property);
}

async function priceRecommendation(property) {
  return await axios.post(`${process.env.REACT_APP_API_URL}/v1/properties/recommend`, property);
}

async function findOptimal(smartSearch) {
  return await axios.get(
    `${process.env.REACT_APP_API_URL}/v1/properties/optimal?=${smartSearch}`
  );
}
